src/fobnail-attester.c

```diff
@@ -129,6 +130,90 @@ static void coap_aik_handler(struct coap_resource_t* resource, struct coap_sessi
 
 }
 
+static inline void free_null_ptr(void *ptr)
+{
+    if (ptr != NULL) {
+        free(ptr);
+        ptr = NULL;
+    }
+}
+
+UsefulBuf encode_meta(struct meta_data *meta)
+{
+    UsefulBuf Buffer = NULLUsefulBuf;
+    QCBOREncodeContext ctx;
+    QCBOREncode_Init(&ctx, Buffer);
```

Comment:
You can't init it with empty buffer... Please check how it is done in `encode_aik()` and the comment mentioning why `_encode_aik()` has to be called twice. This will have to be done in a similar way.

Reply:
Fixed. Please check